src/common/TimeInterval.js

```diff
@@ -16,6 +16,9 @@ class TimeInterval {
 	toString() {
 		if (this.isBounded) {
 			// Using '_' as a separator character since it doesn't appear in ISO dates
+			if (this.startTimestamp === undefined || this.endTimestamp === undefined) {
+				throw Error('startTimestamp or endTimestamp was undefined in a bounded TimeInterval');
```

Comment:
Recommend using console.error here.

Reply:
As long as that's in addition to throwing the error.

Reply:
Can I ask why? What does it get us?

Reply:
Are you asking me why I want the error or John why he wants the console log?

Reply:
Why John wants the console error. Thrown errors get logged if unhandled.